Makefile

```diff
@@ -1,72 +1,101 @@
+default: test
+
+# All builds
+
+build: build-rust build-kotlin
+
+build-rust:
+	cargo build --all
+
+build-kotlin:
+	./gradlew build -x test
+
+build-swift:
+	bin/run-ios-build.sh
+
 build-apk:
 	./gradlew glean-core:build
 	./gradlew glean-sample-app:build
-.PHONY: build-apk
 
-install:
-	$(ANDROID_HOME)/platform-tools/adb install -r ./samples/android/app/build/outputs/apk/debug/glean-sample-app-debug.apk
-.PHONY: install
+.PHONY: build build-rust build-kotlin build-swift build-apk
 
-emulator:
-	$(ANDROID_HOME)/emulator/emulator -avd Nexus_5X_API_P -netdelay none -netspeed full
-.PHONY: install
+# All tests
+
+test: test-rust test-kotlin
+
+test-rust:
+	cargo test --all
 
-clippy: fmt
+test-rust-with-logs:
+	RUST_LOG=glean_core=debug cargo test --all -- --nocapture --test-threads=1
+
+test-kotlin:
+	./gradlew test
+
+test-swift:
+	bin/run-ios-tests.sh
+
+.PHONY: test test-rust test-rust-with-logs test-kotlin test-swift
+
+# Linting
+
+lint: clippy ktlint
+
+clippy:
 	cargo clippy --all --all-targets --all-features -- -D warnings
-.PHONY: lint
 
 ktlint:
 	./gradlew ktlint detekt
-.PHONY: lint
-
-lint: clippy ktlint
-.PHONY: lint
 
 swiftlint:
 	swiftlint --strict
-	swiftformat glean-core/ios samples/ios --swiftversion 5 --verbose --lint
-.PHONY: swiftlint
 
-fmt:
+.PHONY: lint clippy ktlint swiftlint
```

Comment:
For consistency with `build`/`test`/`docs`, maybe name these `lint-kotlin`, `lint-swift`, `lint-rust` etc.?

Reply:
I thought about it ... but stuck with the old names because they are so conveniently named (and so ingrained in my brain to type `cargo clippy`, `swiftlint`, ...)

Reply:
Ok.  Fair enough :)